### PR TITLE
create setup and remove @nn.compact

### LINKED ISSUE #1402: Use `setup` instead of `nn.compact` for Flax
See: https://flax.readthedocs.io/en/latest/design_notes/setup_or_nncompact.html

Edit this file: https://github.com/scverse/scvi-tools/blob/master/scvi/module/_jaxvae.py

### PR DESCRIPTION
**Pull request by mjayasur**
Fixes #1402

### CODE REVIEW

File: scvi/module/_jaxvae.py [hunk 1]
```diff
@@ -60,29 +65,34 @@ class FlaxDecoder(nn.Module):
...
+        self.dense1 = Dense(self.n_hidden)
+        self.dense2 = Dense(self.n_hidden)
+        self.dense3 = Dense(self.n_hidden)
+        self.dense4 = Dense(self.n_input)
+
     def __call__(self, z: jnp.ndarray, batch: jnp.ndarray, is_training: bool):
         disp = self.param(
             "disp", lambda rng, shape: jax.random.normal(rng, shape), (self.n_input, 1)
         )
         is_training = nn.merge_param("is_training", self.is_training, is_training)
```
Comment (adamgayoso) [inline scvi/module/_jaxvae.py:78]: we can get rid of these lines and move all the layers into setup (including batch norm and dropout).

File: scvi/module/_jaxvae.py [hunk 2]
```diff
@@ -104,7 +114,9 @@ class JaxVAE(nn.Module):
     gene_likelihood: str = "nb"
     eps: float = 1e-8
 
-    @nn.compact
+    def setup(self):
+        pass
```
Comment (adamgayoso) [inline scvi/module/_jaxvae.py:118]: let's implement this one too

**Review by adamgayoso — CHANGES_REQUESTED** [review]

**Review by adamgayoso — APPROVED** [review]
